src/dc_msg.rs

```diff
@@ -705,9 +710,11 @@ pub unsafe fn dc_msg_get_text(msg: *const dc_msg_t) -> *mut libc::c_char {
     if msg.is_null() || (*msg).magic != 0x11561156i32 as libc::c_uint {
         return dc_strdup(0 as *const libc::c_char);
     }
-
-    let res = dc_truncate_str(as_str((*msg).text), 30000);
-    res.strdup()
+    if let Some(ref text) = (*msg).text {
```

Comment:
Previously, code panic!'ed if `msg.text` is null. Now, this code returns `NULL`. Probably this is right thing to do.

On other hand, I think that check for `msg.is_null()` is wrong. Function should not silently fail on bogus input, it should panic, imho. I spent some time debugging, while `dc_get_msg_id(NULL) == 0`.

Reply:
All those `is_null` checks are from the legacy C code. It accepted nulls everywhere and proceeded as if nothing happened instead of failing. That is in spirit of C, where `free` accepts NULL.

For Rust code, I am in favor of fail-fast design. Users of the library should avoid passing nulls where data is expected.

Reply:
Quite off-topic, but I disagree that it is in spirit of C. `free` is exception, not rule. Most C API cause UB (e.g SIGSEGV) on NULL pointer: `fopen`, `fprint`, `open`, `readdir`, `regcomp`, ...

On rust code, good. Will add some asserts.

Reply:
> Will add some asserts.

Not in this PR, though. These changes need to be aligned with bindings/android/desktop/ios code, they may rely on this behavior.